lib/src/util/leak_detector_visitor.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.util.leak_detector_visitor;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:linter/src/linter.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+typedef bool DartTypePredicate(DartType type);
+typedef void _VisitVariableDeclaration(VariableDeclaration node);
+typedef bool _Predicate(AstNode node);
+typedef _Predicate _PredicateBuilder(VariableDeclaration v);
+
+abstract class LeakDetectorVisitor extends SimpleAstVisitor {
+  static _PredicateBuilder _isSinkReturn =
```

Comment:
I thought I Had removed all sink references and made names more general, but I overlooked these last references. Changed now.

Regarding the `_PredicateBuilder`s, certainly they are not doing anything in the class other that being referenced from the lists for fields and variables, still I couldn't figure how to convert them in `_Predicate` since I need a function that will be evaluated in the nodes inside the `container` in the `_buildVariableReporter` function and these functions are different for each `VariableDeclaration`, they could be converted to predicates but at the cost of duplication and I am not sure they would be easier to read, I think they would only be a lower level abstraction but not necessarily more readable.

In spite of all said in the previous paragraph, I moved them out of the class since they are kind of pure functions that happen to be used in the class, but don't really depend on it.